x-pack/plugins/security_solution/public/common/components/ml/anomaly/use_anomalies_search.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect, useMemo, useRef } from 'react';
+import { filter, head, noop, orderBy, pipe } from 'lodash/fp';
+import type { MlSummaryJob } from '@kbn/ml-plugin/common';
+import { DEFAULT_ANOMALY_SCORE } from '../../../../../common/constants';
+import * as i18n from './translations';
+import { useUiSetting$ } from '../../../lib/kibana';
+import { useAppToasts } from '../../../hooks/use_app_toasts';
+import { useInstalledSecurityJobs } from '../hooks/use_installed_security_jobs';
+import { notableAnomaliesSearch } from '../api/anomalies_search';
+import type { NotableAnomaliesJobId } from '../../../../overview/components/entity_analytics/anomalies/config';
+import { NOTABLE_ANOMALIES_IDS } from '../../../../overview/components/entity_analytics/anomalies/config';
+import { getAggregatedAnomaliesQuery } from '../../../../overview/components/entity_analytics/anomalies/query';
+import type { inputsModel } from '../../../store';
+import { isJobFailed, isJobStarted } from '../../../../../common/machine_learning/helpers';
+
+export enum AnomalyJobStatus {
+  'enabled',
+  'disabled',
+  'uninstalled',
+  'failed',
+}
+
+export interface AnomaliesCount {
+  name: NotableAnomaliesJobId;
+  jobId?: string;
+  count: number;
+  status: AnomalyJobStatus;
+}
+
+interface UseNotableAnomaliesSearchProps {
+  skip: boolean;
+  from: string;
+  to: string;
+}
+
+export const useNotableAnomaliesSearch = ({
+  skip,
+  from,
+  to,
+}: UseNotableAnomaliesSearchProps): {
+  isLoading: boolean;
+  data: AnomaliesCount[];
+  refetch: inputsModel.Refetch;
+} => {
+  const [data, setData] = useState<AnomaliesCount[]>(formatResultData([], []));
+  const refetch = useRef<inputsModel.Refetch>(noop);
+
+  const {
+    loading: installedJobsLoading,
+    isMlUser,
+    jobs: installedSecurityJobs,
+  } = useInstalledSecurityJobs();
+  const [loading, setLoading] = useState(true);
+  const { addError } = useAppToasts();
+  const [anomalyScoreThreshhold] = useUiSetting$<number>(DEFAULT_ANOMALY_SCORE);
+
+  const { notableAnomaliesJobs, query } = useMemo(() => {
+    const newNotableAnomaliesJobs = installedSecurityJobs.filter(({ id }) =>
+      NOTABLE_ANOMALIES_IDS.some((notableJobId) => matchJobId(id, notableJobId))
+    );
+
+    const newQuery = getAggregatedAnomaliesQuery({
+      jobIds: newNotableAnomaliesJobs.map(({ id }) => id),
+      anomalyScoreThreshhold: anomalyScoreThreshhold ?? 50,
```

Comment:
Do we really need to set the default value again? `50` is already the default value of the DEFAULT_ANOMALY_SCORE setting

Reply:
We don't need it. I added it only because typescript was complaining that it could be undefined. But now I removed ` ?? 50` from the code, and it isn't complaining. 🤔 
Let me delete it 👍